src/main/java/com/example/scholar_gql_poc/resolver/ScholarQueryResolver.java

```diff
@@ -20,4 +21,13 @@ public ScholarQueryResolver(ScholarRepository scholarRepository) {
    public List<Scholar> getAllScholars() {
         return scholarRepository.findAll();
     }
+
+    public Scholar getScholarById(String id)
+    {
+        Scholar scholar = null;
+        if(scholarRepository.existsById(id)) {
+            return scholarRepository.findById(id).get();
+        }
+        return scholar;
+    }
```

Comment:
Catch the exception in case there is no result for a particular scholar and return not found massage

Reply:
Addressed your comments.